### PR TITLE
Fix visual progress bar bug / properly reset progress bar

### LINKED ISSUE #4567: validation progress bar does not update correctly
Does not show progress. Reason is line 359 in progress.py:

`self.val_progress_bar.total = convert_inf(self.total_val_batches)`

should be:

`self.val_progress_bar.reset(self.total_val_batches)`

Same applies to other places where progress bar total set.

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #4567

A visual bug occurs at the very initial state of the progress bar:

```
Epoch 0:  33%|███▎      | 5/15 [00:05<00:10,  1.01s/it, loss=0.892, v_num=63]
Validating: 0it [00:00, ?it/s]
```

It shows "0 it" and the total is not displayed. But we can make it show just 0% like so:

```
Epoch 0:  33%|███▎      | 5/15 [00:05<00:10,  1.01s/it, loss=1.57, v_num=64] 
Validating:   0%|          | 0/10 [00:00<?, ?it/s]
```

by calling the `tqdm.reset()` method instead of manually setting the total amount.

This is very hard to spot, because the progress bar usually updates very fast. You have to add a `time.sleep` to the training loop to observe it.

### CODE REVIEW

**Review by tchaton — APPROVED** [review]

File: pytorch_lightning/callbacks/progress.py [hunk 1]
```diff
@@ -387,8 +388,14 @@ def _update_bar(self, bar):
...
 
-def convert_inf(x):
+def convert_inf(x: Union[int, float]) -> Optional[int]:
     """ The tqdm doesn't support inf values. We have to convert it to None. """
     if x == float('inf'):
         return None
     return x
+
+
+def reset(bar: tqdm, total: Optional[int]) -> None:
```
Comment (carmocca) [inline pytorch_lightning/callbacks/progress.py:398]:
```suggestion
def reset(bar: tqdm, total: Optional[int] = None) -> None:
```

File: pytorch_lightning/callbacks/progress.py [hunk 2]
```diff
@@ -387,8 +388,14 @@ def _update_bar(self, bar):
             bar.update(delta)
 
 
-def convert_inf(x):
+def convert_inf(x: Union[int, float]) -> Optional[int]:
```
Comment (carmocca) [inline pytorch_lightning/callbacks/progress.py:391]:
```suggestion
def convert_inf(x: Optional[Any]) -> Optional[Any]:
```

Comment (carmocca) [reply pytorch_lightning/callbacks/progress.py:391]: You could make an argument that you want to tighten the definition to the expected use (not the allowed). But in that case, it'd be

```suggestion
def convert_inf(x: Optional[Union[int, float]]) -> Optional[Union[int, float]]:
```

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/callbacks/progress.py (hunk 2 above)
Comment (awaelchli) [reply pytorch_lightning/callbacks/progress.py]: ok, I think I need to add imports

**Comment by Borda** [comment]
would be fine to merge it after #5495

**Review by Borda — APPROVED** [review]

**Pull request merged**